Add in-app purchase scanning support to precheck

taquitos opened the pull request:
IAP sometimes fails review if you include the word "free" in your description. That's silly. Oh well, let's go ahead and scan all our IAP to make sure we're not going to get rejected!

- Add option to disable in-app purchase scanning
- Add rule to check for the word "free" in IAP

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid

Commit:
Add in-app purchase scanning support
Add option to disable in-app purchase scanning
Add rule to check for the word "free" in IAP

harlanhaskins commented on precheck/lib/precheck/rule_processor.rb:
```diff
@@ -175,6 +176,30 @@ def self.generate_text_items_to_check(app: nil, app_version: nil)
+
+      should_include_iap = Precheck.config[:include_in_app_purchases]
+      if should_include_iap
+        UI.message "Reading in-app purchases. If you have alot, this might take a while"
```
Nitpick: `a lot`

taquitos committed:
Update rule_processor.rb

Lulz, alot (reminds me of http://hyperboleandahalf.blogspot.com/2010/04/alot-is-better-than-you-at-everything.html)

taquitos commented:
@harlanhaskins 🤣  http://hyperboleandahalf.blogspot.com/2010/04/alot-is-better-than-you-at-everything.html

ohayon commented on precheck/lib/precheck/rule_processor.rb:
```diff
@@ -175,10 +176,34 @@ def self.generate_text_items_to_check(app: nil, app_version: nil)
+      should_include_iap = Precheck.config[:include_in_app_purchases]
+      if should_include_iap
+        UI.message "Reading in-app purchases. If you have a lot, this might take a while"
+        UI.message "You can disable IAP checking by adding the include_in_app_purchases flag"
```
Should this say "...checking by setting the `include_in_app_purchases` flag to `false`"

ohayon commented on precheck/lib/precheck/rule_processor.rb:
```diff
@@ -175,10 +176,34 @@ def self.generate_text_items_to_check(app: nil, app_version: nil)
+        in_app_purchases = app.in_app_purchases.all
+        in_app_purchases ||= []
+        in_app_purchases.each do |purchase|
+          items += self.collect_iap_language_items(purchase_edit_versions: purchase.edit.versions)
```
Take this or feel free to leave it: I don't believe that `self` is necessary here, but sometimes its nice obvs.

ohayon reviewed (approved):
👍 💯 🎉

taquitos committed:
Update rule_processor.rb

Cleaned up message about setting the `include_in_app_purchases` flag
removed call to `self.` to make code more consistent

KrauseFx commented:
pls delete branches :P

revolter commented:
But there's no way to pass it via `deliver`, right?
